### PR TITLE
docs: rework CODEOWNERS into a list of Functional Areas & Owners

### PR DESCRIPTION
**Pull request by bajtos**
This year, I set a goal for myself to find a way how to spend less time on reviewing code and have more time for implementing new features and other activities pushing the framework further. After discussing the idea with @dhmlau, we decided that the first step is to enable other core maintainers (especially in @strongloop/sq-lb-apex) to take on more responsibility & ownership of various parts of the codebase (the functional areas).

We have discussed the functional areas and new owners internally within the IBM team first. Now in this pull request, I am reworking the content of CODEOWNERS file to keep track of the Functional Areas and their Owners we identified in our discussion. I expect the list of owners to evolve over time, it's not necessary to get it perfect on the first try.

Obviously, promoting a person to a Functional Area Owner is not enough to make them successful in their new role. In the next weeks and months, I'd like to help them to learn more about their functional area, so that eventually most of day-to-day conversation won't require my input.

In this initial version, I am not adding any non-IBMers from @strongloop/loopback-maintainers team yet, because I don't know if any of you would like to be listed as a code owner and I don't want you to feel like we are imposing on you more work and responsibilities than you are willing to take. Having said that, if you would like to take more ownership, then let us know - preferably by commenting here in this pull request or opening a follow-up pull request adding you to the owners list for one or more functional areas.

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

**Comment by dhmlau** [comment]
Reminder: once this is landed, we probably need to update https://github.com/strongloop/loopback4-example-shopping/blob/master/CODEOWNERS.

File: CODEOWNERS [hunk 1]
```diff
@@ -1,50 +1,356 @@
...
+#    of incoming pull requests.
+#
 # Lines starting with '#' are comments.
 # Each line is a file pattern followed by one or more owners,
 # the last matching pattern has the most precedence.
 #
 # See https://help.github.com/articles/about-codeowners/
+#
+# IMPORTANT: Because the last matching pattern takes the most precedence,
+# we need to list functiona areas from most generic (e.g. Models &
```
Comment (nabdelgadir) [inline CODEOWNERS]: `functional`

Comment (dhmlau) [reply CODEOWNERS]: maybe `list the functional areas..`?

**Comment by bajtos** [comment]
I am going to land this pull request to avoid merge conflicts. I am happy to improve CODEOWNERS further if there is any more feedback on the current version.

**Comment by bajtos** [comment]
> Reminder: once this is landed, we probably need to update https://github.com/strongloop/loopback4-example-shopping/blob/master/CODEOWNERS.

See https://github.com/strongloop/loopback4-example-shopping/pull/534

I think we should eventually update CODEOWNERS in juggler & connector repositories too.

**Pull request merged**